### ISSUE TITLE
CNN rnn hybrid

### ISSUE THREAD

**Issue by aryansingh920** [aryansingh920/image-captcha-recognition-scalable_project2_group24#4]
def 

create_hybrid_cnn_rnn_model(captcha_length, captcha_num_symbols, input_shape, cnn_depth=5, module_size=2, rnn_units=128):

    input_tensor = keras.Input(input_shape)

    # CNN Layers
    x = input_tensor
    for i, module_length in enumerate([module_size] * cnn_depth):
        for j in range(module_length):
            x = keras.layers.Conv2D(32 * 2 ** min(i, 3), kernel_size=3, padding='same',
                                    kernel_initializer='he_uniform')(x)
            x = keras.layers.BatchNormalization()(x)
            x = keras.layers.Activation('relu')(x)
        x = keras.layers.MaxPooling2D(2)(x)

    # Reshape to (batch_size, timesteps, features) for RNN input
    x = keras.layers.Reshape((-1, x.shape[-1] * x.shape[-2]))(x)  # Flatten the spatial dimensions into a feature vector

    # RNN Layers
    x = keras.layers.LSTM(rnn_units, return_sequences=True)(x)

    # Output layer for each character in the CAPTCHA
    outputs = [keras.layers.Dense(captcha_num_symbols, activation='softmax', name='char_%d' % (i + 1))(x[:, i, :]) 
               for i in range(captcha_length)]

    model = keras.Model(inputs=input_tensor, outputs=outputs)
    return model
